app/controllers/api/v1/messages_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class MessagesController < Api::V1::BaseController
+      before_action only: %i[create] do
+        require_access_token %w[read write]
+      end
+
+      before_action only: %i[index] do
+        require_access_token %w[read]
+      end
+
+      rescue_from ActiveRecord::RecordNotFound do
+        render(
+            json: {error: I18n.t("conversations.not_found")},
+            status: :not_found
+        )
+      end
+
+      def create
+        conversation = conversation_service.find!(params[:conversation_id])
+        opts = params.require(:body)
+        message = current_user.build_message(conversation, text: opts[:body])
+        message.save!
+        Diaspora::Federation::Dispatcher.defer_dispatch(current_user, message)
+        render json: message_json(message), status: :created
+      end
+
+      def index
+        conversation = conversation_service.find!(params[:conversation_id])
+        conversation.set_read(current_user)
+        render(
+            json: conversation.messages.map {|x| message_json(x) },
```

Comment:
- Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
- Layout/FirstParameterIndentation: Indent the first parameter one step more than the start of the previous line.